packages/@react-stately/combobox/src/useComboBoxState.ts

```diff
@@ -153,7 +183,15 @@ export function useComboBoxState<T extends object>(props: ComboBoxStateProps<T>)
     } else {
       lastSelectedKey.current = null;
       setSelectedKey(null);
-      shouldClose = selectedKey === null;
+
```

Comment:
https://github.com/adobe/react-spectrum/pull/1449/files#diff-52d3ae5861fe9f80aaec78778d0a666b2b9d10dbf2aac7133bd6d3d70cade0faR180-R182

Just noticed that it won't close when clicking out of the combobox, perhaps this should be
```
 if (!allowsCustomValue) {
      resetInputValue();
      // Also close if the reset input value is empty
      shouldClose = inputValue === lastValue.current || lastValue.current === '';
```
or perhaps blur should always close the menu

Reply:
When would blur not close the menu?

Reply:
yep, I couldn't think of a scenario where blur wouldn't close the menu so I made the above change